modules/nomad-module/main.tf

```diff
@@ -0,0 +1,124 @@
+terraform {
+  required_providers {
+    grid = {
+      source = "threefoldtech/grid"
+    }
+  }
+}
+
+locals {
+  servers_count = length(var.servers) + 1
+  server_flist  = "https://hub.grid.tf/aelawady.3bot/abdulrahmanelawady-nomad-server-latest.flist"
+  client_flist  = "https://hub.grid.tf/aelawady.3bot/abdulrahmanelawady-nomad-client-latest.flist"
+  entrypoint    = "/sbin/zinit init"
+}
+
+resource "grid_network" "net" {
+  nodes       = var.network.nodes
+  ip_range    = var.network.ip_range
+  name        = var.network.name
+  description = var.network.description
+}
+
+resource "grid_deployment" "server1" {
+  node         = var.servers[0].node
+  network_name = grid_network.net.name
+
+  vms {
+    name       = var.servers[0].name
+    flist      = local.server_flist
+    cpu        = var.servers[0].cpu
+    memory     = var.servers[0].memory
+    ip         = var.first_server_ip
+    publicip   = var.servers[0].publicip
+    publicip6  = var.servers[0].publicip6
+    planetary  = var.servers[0].planetary
+    entrypoint = local.entrypoint
+    mounts {
+      disk_name   = var.servers[0].disk.name
+      mount_point = var.servers[0].mount_point
+    }
+    env_vars = {
+      SSH_KEY       = var.ssh_key
+      NOMAD_SERVERS = local.servers_count
+    }
+  }
+
+  disks {
+    name = var.servers[0].disk.name
+    size = var.servers[0].disk.size
+  }
+}
+
+resource "grid_deployment" "servers" {
```

Comment:
I don't get why servers are separated into two deployments while they can be on the same one. The only difference between the first server and the rest is is just one environment variable which you can use ternary operator to set it if the index is 0